centreon/www/class/centreonACL.class.php

```diff
@@ -45,7 +46,7 @@ class CentreonACL
     const ACL_ACCESS_READ_ONLY = 2;
 
     private $userID; /* ID of the user */
-    private $parentTemplates = null;
+    private ?array $parentTemplates = null;
```

Comment:
Would it be relevant to remove the nullable type and call `$this->loadParentTemplates()` in the __construct ?

Thus you wouldn't have to check if the value is null in each method using it.

Reply:
Yes, it's a possibility, but in doing so I'd have to make sure it didn't impact anything else, and since it's unlikely that this scope would be tested, I'd prefer to keep the impact to a minimum.